File: tags/REL-1.0/src/Cliente/Cliente/Validaciones/SoloNumeros.cs
using System.Windows.Controls;
using System.Text.RegularExpressions;

namespace Trascend.Bolet.Cliente.Validaciones
{
    public class SoloNumeros : ValidationRule
    {
        private const string _expresionRegular = "\\d";

        public override ValidationResult Validate(object value, System.Globalization.CultureInfo cultureInfo)
        {
            string valor = (string) value;

            Regex regular = new Regex(_expresionRegular);

            if (!regular.IsMatch(valor) && !string.IsNullOrEmpty(valor))
            {
               return new ValidationResult(false, Recursos.MensajesValidaciones.SoloNumeros);
            }
            else
                return ValidationResult.ValidResult;
        }
    }
}

File: trunk/src/Cliente/Cliente_Fac/Validaciones/SoloNumeros.vb
Imports System.Windows.Controls
Imports System.Text.RegularExpressions

Namespace Validaciones
    Public Class SoloNumeros
        Inherits ValidationRule
        Private Const _expresionRegular As String = "\d"

        Public Overrides Function Validate(ByVal value As Object, ByVal cultureInfo As System.Globalization.CultureInfo) As ValidationResult
            Dim valor As String = DirectCast(value, String)

            Dim regular As New Regex(_expresionRegular)
            If (valor <> Nothing) Then

                If Not regular.IsMatch(valor) AndAlso Not String.IsNullOrEmpty(valor) Then
                    Return New ValidationResult(False, Recursos.MensajesValidaciones.SoloNumeros)
                Else
                    Return ValidationResult.ValidResult
                End If
            Else
                Return ValidationResult.ValidResult
            End If
        End Function
    End Class
End Namespace